internal/tokenutil/jwtmiddleware.go

```diff
@@ -0,0 +1,40 @@
+package tokenutil
```

Comment:
consider functionality and packaging from top to down. Meaning from global to smaller scales. In this particular example we have JWT middleware, which is type of middleware. So instead of adding multiple packages for each middleware as the root package it would be preferential to have a `middleware` package with `jwt.go` which would contain our `JWTAuth` middleware function. Later on we can expand our middleware package with other middlewares, such as request logging, metrics, etc. easily and without much alteration to the existing code.

Reply:
got it, but `internal/middleware` is correct, right?

Reply:
That would be better, yes.

Reply:
done